File: problem29.rb
def distinct_powers
  bases = (2..100).to_a
  exponent = (2..100).to_a
  terms = []
  bases.each do |base|
    exponent.each do |exp|
      terms << base ** exp    
    end
  end

  terms.uniq.length

end

p distinct_powers

File: problem29.py
def distinct_powers():
  bases = range(2,101)
  exponents = range(2,101)
  result = []
  for base in bases:
    for exponent in exponents:
      value = base ** exponent
      if (value in result):
        continue
      else:
        result.append(base ** exponent)
  return len(result)
print(distinct_powers())
